expression/builtin_like.go

```diff
@@ -89,8 +89,11 @@ func (b *builtinLikeSig) evalInt(row chunk.Row) (int64, bool, error) {
 			b.isMemorizedPattern = true
 		}
 	}
+	// should be thread-safe
 	if !b.isMemorizedPattern {
-		b.pattern.Compile(patternStr, byte(escape))
+		pattern := b.collator().Pattern()
+		pattern.Compile(patternStr, byte(escape))
+		return boolToInt64(pattern.DoMatch(valStr)), false, nil
```

Comment:
Could we move the `ConstItem` judgment to `getFunction`? So that we can keep this optimization in both vectorized and non-vectorized version and avoid data race.

Reply:
we can refine the expression system in terms of this later.

Reply:
It's not a hard change. For both correctness and performance, we may do it here.

Reply:
if both args[1] and args[2] are const items, the `b.pattern` could be set in `getFunction`, or a data race may happen at line 86.